lib/livebook_web/live/app_session_live.ex

```diff
@@ -147,29 +147,53 @@ defmodule LivebookWeb.AppSessionLive do
               session_pid={@session.pid}
               client_id={@client_id}
               cell_id={output_view.cell_id}
-              input_values={output_view.input_values}
+              input_views={output_view.input_views}
             />
           </div>
           <%= if @data_view.app_status.execution == :error do %>
-            <div class="error-box flex items-center justify-between">
-              <div class="flex items-center gap-2">
-                <.remix_icon icon="error-warning-line" class="text-xl" />
-                <span>Something went wrong</span>
+            <div class={[
+              "flex justify-between items-center px-4 py-2 border-l-4 shadow-custom-1",
+              "text-red-400 border-red-400"
+            ]}>
+              <div>
+                Something went wrong
+              </div>
+              <div class="flex items-center gap-6">
+                <span class="tooltip top" data-tooltip="Debug">
+                  <.link
+                    :if={@livebook_authenticated?}
+                    navigate={~p"/sessions/#{@session.id}" <> "#cell-#{@data_view.errored_cell_id}"}
+                  >
+                    <.remix_icon icon="terminal-line" />
+                  </.link>
+                </span>
+                <button
+                  class={[
+                    "button-base bg-transparent",
+                    "border-red-400 text-red-400 hover:bg-red-50 focus:bg-red-50"
+                  ]}
+                  phx-click="queue_errored_cells_evaluation"
+                >
+                  <.remix_icon icon="play-circle-fill" class="align-middle mr-1" />
+                  <span>Retry</span>
+                </button>
               </div>
-              <.link
-                :if={@livebook_authenticated?}
-                navigate={~p"/sessions/#{@session.id}" <> "#cell-#{@data_view.errored_cell_id}"}
-              >
-                <span>Debug</span>
-                <.remix_icon icon="arrow-right-line" />
-              </.link>
             </div>
           <% end %>
         </div>
         <div style="height: 80vh"></div>
       </div>
-      <div :if={show_app_status?(@data_view.app_status)} class="fixed right-6 bottom-4">
-        <.app_status status={@data_view.app_status} />
+      <div class="fixed right-3 bottom-4 flex flex-col gap-2 items-center text-gray-600 w-10">
+        <span
+          :if={@data_view.app_status.execution != :executing and @data_view.any_stale?}
+          class="tooltip left"
+          data-tooltip="Process changes"
```

Comment:
```suggestion
          data-tooltip="Some inputs have changed since they were last processed. Click this button to reprocess with latest values"
```